### PR TITLE
all tests passing

### PR DESCRIPTION
**Pull request by invegat**

### CODE REVIEW

File: TopologicalSort/topological_sort.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            return []
+        for k in empty_keys:
+            if k in jobs:
+                jobs.remove(k)
+                r.append(k)
+            del ld[k]
+
+            for i, (key, value) in enumerate(ld.items()):
+                if k in value:
+                    ld[key].remove(k)
```
Comment (BrannanC) [inline TopologicalSort/topological_sort.py]: You could initialize empty_keys outside the while loop and do the empty_keys logic here so there's no need to iterate over ld each time to get items with no dependencies.

File: TopologicalSort/topological_sort.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#
+# topological_sort(tasks, deps) = [1,2,4,3] or [2,1,4,3]
+#
+# Author: BrannanC
+# """
+from collections import defaultdict
+
+
+def topological_sort(j, deps):
+    jobs = j[::1]
```
Comment (BrannanC) [inline TopologicalSort/topological_sort.py:25]: Is this just to copy j so you don't mutate it?

**Review by BrannanC — COMMENTED** [review]
Great job! 🎉
This is definitely a harder challenge!
In the future please clean up the unnecessary comments so it's easier to read.

File: TopologicalSort/topological_sort.py (hunk 2 above)
Comment (invegat) [reply TopologicalSort/topological_sort.py:25]: yes

File: TopologicalSort/topological_sort.py (hunk 1 above)
Comment (invegat) [reply TopologicalSort/topological_sort.py]: I can make new empty keys with the  ``ld[key].remove(k)``  if k is the last value list member, I delete the old empty keys.   If no new empty key is made return the empty list.

**Pull request merged**